tools/windows-node-installer/pkg/cloudprovider/aws/ec2_test.go

```diff
@@ -37,3 +44,226 @@ func TestRSADecrypt(t *testing.T) {
 		t.Fatalf("Error decrypting message expected %v got %v", msg, decryptedMsg)
 	}
 }
+
+type byPort []string
+
+// TestGetMissingRules tests the getMissingRules function which gets all the missing rules for local IPs
+func TestGetMissingRules(t *testing.T) {
+	myIP := "144.121.20.163"
+	otherIP := "100.121.20.163"
+	vpcCidr := "1.1.1.1/32"
+	testIO := []struct {
+		name     string
+		in       []*ec2.IpPermission
+		expected []*ec2.IpPermission
+	}{
+		//test with an empty input of rules
+		{
+			name: "Empty input",
+			in:   []*ec2.IpPermission{},
+			expected: []*ec2.IpPermission{
+				(&ec2.IpPermission{}).
+					// -1 is to allow all ports.
+					SetIpProtocol("-1").
+					SetIpRanges([]*ec2.IpRange{
+						(&ec2.IpRange{}).
+							SetCidrIp(vpcCidr),
+					}),
+				(&ec2.IpPermission{}).
+					SetIpProtocol("tcp").
+					SetFromPort(WINRM_PORT).
+					SetToPort(WINRM_PORT).
+					SetIpRanges([]*ec2.IpRange{
+						(&ec2.IpRange{}).
+							SetCidrIp(myIP + "/32"),
+					}),
```

Comment:
Please add a helper for this and call it in all places instead of repeating the code.